waiter/src/waiter/process_request.clj

```diff
@@ -180,17 +183,121 @@
     (deliver reservation-status-promise promise-value)
     (utils/exception->response (ex-info message (assoc metrics-map :status status) error) request)))
 
+(defn make-stream-reader-executor
+  "Creates and returns the ThreadPoolExecutor used for reading the input stream."
+  [concurrency-level keep-alive-mins]
+  (let [queue (LinkedBlockingQueue.)
```

Comment:
Should we fix the size of the `LinkedBlockingQueue`? Unbounded could be a problem :)

Reply:
We will have at most one entry in the queue per queued/in-flight request. I feel like we do not need a bound here, the request queue bounds can handle the limiting.

Reply:
Actually, let me add a reasonably high bound for safety.